src/engine/Map.cxx

```diff
@@ -434,17 +434,17 @@ Point Map::findNodeInMap(const SDL_Point &screenCoordinates) const
     // include 2 columns on each side, since calculated values can be that far off
     for (int i = 0; i <= 2; i++)
     {
-      if (isClickWithinTile(screenCoordinates, isoX, isoY) &&
+      if (isClickWithinTile(screenCoordinates, isoX, isoY, layer) &&
           (foundCoordinates.z < mapNodes[isoX * m_columns + isoY]->getCoordinates().z))
       {
         foundCoordinates = mapNodes[isoX * m_columns + isoY]->getCoordinates();
       }
-      if (isClickWithinTile(screenCoordinates, isoX + i, isoY) &&
+      if (isClickWithinTile(screenCoordinates, isoX + i, isoY, layer) &&
           (foundCoordinates.z < mapNodes[(isoX + i) * m_columns + isoY]->getCoordinates().z))
       {
         foundCoordinates = mapNodes[(isoX + i) * m_columns + isoY]->getCoordinates();
       }
-      if (isClickWithinTile(screenCoordinates, isoX, isoY - i) &&
+      if (isClickWithinTile(screenCoordinates, isoX, isoY - i, layer) &&
           (foundCoordinates.z < mapNodes[isoX * m_columns + (isoY - i)]->getCoordinates().z))
       {
         foundCoordinates = mapNodes[isoX * m_columns + (isoY - i)]->getCoordinates();
```

Comment:
Consider to replace this implementation with something like this:

`   

     std::set<std::tuple<int, int>> mapPoints{std::make_tuple(isoX, isoY)};

    // find additional coordinates, include 2 columns on each side, since calculated values can be that far off
    for (int i = 1; i <= 2; i++)
    {
      mapPoints.insert(std::make_tuple(isoX + i, isoY));
      mapPoints.insert(std::make_tuple(isoX, isoY - 1));
    }

    for (auto [x, y] : mapPoints)
    {
      if (isClickWithinTile(screenCoordinates, x, y, layer) &&
          (foundCoordinates.z < mapNodes[x * m_columns + y]->getCoordinates().z))
      {
        foundCoordinates = mapNodes[x * m_columns + y]->getCoordinates();
        break;
      }
    }
`

Reply:
I think we should consult @JimmySnails  regarding that.
I think that it does not have "else if" on purpose.

Reply:
resolved